services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -0,0 +1,878 @@
+
+        when: "Try to swap the inactive flow"
+        northbound.swapFlowPath(flow.id)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not swap paths: Protected flow path $flow.id is not in ACTIVE state"
+
+        and: "Restore topology, delete flows and reset costs"
+        //TODO investigate I think I have choosen switch pair incorrectly
+        // possibly i have to choose switch pair without a-switch
+        northbound.portUp(currentIsls[0].srcSwitch.dpId, currentIsls[0].srcPort)
+        northbound.portUp(currentIsls[0].dstSwitch.dpId, currentIsls[0].dstPort)
+        Wrappers.wait(WAIT_OFFSET * 2) { assert northbound.getFlowStatus(flow.id).status == FlowState.UP }
+        northbound.portUp(protectedIsls[0].srcSwitch.dpId, protectedIsls[0].srcPort)
+        northbound.portUp(protectedIsls[0].dstSwitch.dpId, protectedIsls[0].dstPort)
+        broughtDownPorts.every { northbound.portUp(it.switchId, it.portNo) }
+        flowHelper.deleteFlow(flow.id)
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            northbound.getAllLinks().each { assert it.state != IslChangeType.FAILED }
+        }
+        database.resetCosts()
+    }
+
+    @Unroll
+    def "#flowDescription flow is DOWN when protected and alternative paths are not available"() {
+        given: "Two active neighboring switches with two not overlapping paths at least"
+        def (Switch srcSwitch, Switch dstSwitch) = getNeighboringSwitchPair(2)
+
+        and: "A flow with protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.allocateProtectedPath = true
+        flow.maximumBandwidth = bandwidth
+        flowHelper.addFlow(flow)
+        def flowInfoPath = northbound.getFlowPath(flow.id)
+        assert flowInfoPath.protectedPath
+
+        when: "All alternative paths are unavailable (bring ports down on the source switch and on the protected path)"
+        def currentPath = pathHelper.convert(flowInfoPath)
+        def currentProtectedPath = pathHelper.convert(flowInfoPath.protectedPath)
+        List<PathNode> broughtDownPorts = []
+        database.getPaths(srcSwitch.dpId, dstSwitch.dpId)*.path.findAll {
+            it != pathHelper.convert(northbound.getFlowPath(flow.id))
+        }.unique {
+            it.first()
+        }.each { path ->
+            def src = path.first()
+            broughtDownPorts.add(src)
+            northbound.portDown(src.switchId, src.portNo)
+        }
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.FAILED
+            }.size() == broughtDownPorts.size() * 2
+        }
+
+        then: "Protected path is not recalculated because all alt paths are unavailable"
```

Comment:
I am not sure about this step. It looks like it is redundant.

Reply:
as you wish)